tests/ShadowDOM/js/Range.js

```diff
@@ -11,76 +11,525 @@
 suite('Range', function() {
 
   var wrap = ShadowDOMPolyfill.wrap;
+  var unwrap = ShadowDOMPolyfill.unwrap;
+  var wrapIfNeeded = ShadowDOMPolyfill.wrapIfNeeded;
+  var isNativeShadowDomSupported;
 
-  var div;
+  var hosts;
+  var customElementPrefix = "range-custom-element-";
+  var nativeCustomElementPrefix = "range-native-element-";
+  var customElementIndex=0;
+  var nativeCustomElementIndex=0;
 
-  teardown(function() {
-    if (div && div.parentNode)
-      div.parentNode.removeChild(div);
-    div = undefined;
+  setup(function() {
+    isNativeShadowDomSupported = !!unwrap(document.createElement('div')).createShadowRoot;
```

Comment:
Actually I already started working on the document.execCommand and I found my self needing some of the code in the Range tests, so I moved those functions to a ShadowDomTestUtils.js to allow me to create custom and standard elements with different shadow dom setup native and polyfill mode. I was also thinking other unit tests can use those functions to setup the tests and can test dynamically what the result of the native implementation. 

All the changes you suggested are checked in.